File: getpoem.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <assert.h>
int main(int argc, char **argv){
    int peek = strcmp(argv[1], "false");
    char * host = argv[2];
    int git = argc > 3;
    char * sha = git ? argv[3] : 0;

    int idx;
    
    FILE *poemfile = fopen("/home/hmrm/.oh-my-zsh/custom/poems.txt", "r");
    FILE *indexfile = fopen("/home/hmrm/.oh-my-zsh/custom/.poem_line_tmp.tmp", "r");
    if(!indexfile){
        indexfile = fopen("/home/hmrm/.oh-my-zsh/custom/.poem_line_tmp.tmp", "w");
        idx = 0;
        fwrite(&idx, sizeof(int), 1, indexfile);
    } else {
        fread(&idx, sizeof(int), 1, indexfile);
    }
    
    char * buffer = NULL;
    size_t n = 0;
    for (int i = 0; i <= idx; i++){
        int ret = getline(&buffer, &n, poemfile);
        if (ret <= 0) {
            fclose(indexfile);
            indexfile = fopen("/home/hmrm/.oh-my-zsh/custom/.poem_line_tmp.tmp", "w") ;
            idx = 0;
            fwrite(&idx, sizeof(int), 1, indexfile);
            fclose(indexfile);
            fclose(poemfile);
            exit(0);
        }
    }
    
    printf("%s", buffer);

    if(!peek){
        idx++;
        fclose(indexfile);
        indexfile = fopen("/home/hmrm/.oh-my-zsh/custom/.poem_line_tmp.tmp", "w") ;
        fwrite(&idx, sizeof(int), 1, indexfile);
    }
    fclose(indexfile);
    fclose(poemfile);
}

File: getpoem.py
import sys
peek = sys.argv[1]
host = sys.argv[2]
git = False
if(len(sys.argv) >= 4):
    git = True
    sha = int(sys.argv[3], 16)

with open("/home/hmrm/.oh-my-zsh/custom/poems.txt") as f:
   lines = f.readlines(4)

read = ''
try:
    with open("/home/hmrm/.oh-my-zsh/custom/.poem_line_tmp.tmp", "r") as tmp:
        read=tmp.read().strip()
except IOError:
    pass
        
if read == '':
    read = '0'

index = int(read)
if not peek == "true":
    with open("/home/hmrm/.oh-my-zsh/custom/.poem_line_tmp.tmp", "w") as tmp:
        tmp.write(str(index + 1))

print lines[index % len(lines)]